azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -452,6 +453,40 @@ func resourceArmStorageAccount() *schema.Resource {
 				},
 			},
 
+			"blob_account_properties": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"static_website": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+									"index_document": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"error_document_404_path": {
```

Comment:
Why here a path postfix? In the azure cli it is 404-document and index-document. See https://docs.microsoft.com/de-de/azure/storage/blobs/storage-blob-static-website-how-to?tabs=azure-cli

Reply:
I named these arguments based on the names from `giovanni`. 

https://github.com/tombuildsstuff/giovanni/blob/89b9b60252d146d1412bb2cbefa1183c30f4c9c0/storage/2018-11-09/blob/accounts/models.go#L20-L27

It's also worth noting in Azure PowerShell, the input is called `ErrorDocument404Path` as well.

https://docs.microsoft.com/en-us/powershell/module/az.storage/enable-azstoragestaticwebsite?view=azps-3.4.0